sql/sql_parse.cc

```diff
@@ -1287,12 +1295,18 @@ bool do_command(THD *thd)
 #ifdef WITH_WSREP
   DEBUG_SYNC(thd, "wsrep_before_before_command");
   /*
-    Aborted by background rollbacker thread.
-    Handle error here and jump straight to out
+    If this command does not return a result, then we
+    instruct wsrep_before_command() to skip result handling.
+    This causes BF aborted transaction to roll back but keep
+    the error state until next command which is able to return
+    a result to the client.
   */
-  if (wsrep_before_command(thd))
+  if (wsrep_before_command(thd, wsrep_command_no_result(command)))
```

Comment:
Perhaps comment here would be in place that `thd->store_globals()` is called internally from `wsrep_before_command()` if it was not called yet.

Reply:
Added a comment about thd->store_globals()